source/MaterialXCore/Version.cpp

```diff
@@ -1148,9 +1155,10 @@ void Document::upgradeVersion()
                     string channelString = channelsInput ? channelsInput->getValueString() : EMPTY_STRING;
                     size_t sourceChannelCount = CHANNEL_COUNT_MAP.at(inInput->getType());
                     size_t destChannelCount = CHANNEL_COUNT_MAP.at(node->getType());
-                    if (inInput->hasValue())
+                    const bool isDisconnected = inInput->getNodeName().empty() && inInput->getNodeGraphString().empty() && inInput->getInterfaceName().empty();
```

Comment:
Here I'd recommend naming the boolean `hasConnection` for consistency, and ideally we should use any available helper methods (e.g. `hasInterfaceName`) to write this logic more clearly.